### ISSUE TITLE
A null comment in security integration (external oauth) throws a parsing error.

### ISSUE THREAD

**Issue by rahulj51** [Snowflake-Labs/terraform-provider-snowflake#1755]
**Provider Version**

0.62.0

**Terraform Version**

1.4.5

**Describe the bug**

An existing security integration with a null comment throws a parsing error. 

```
2023-04-25T18:04:59.412+0530 [INFO]  provider.terraform-provider-snowflake_v0.62.0: 2023/04/25 18:04:59 [DEBUG] sql-conn-query [query SHOW SECURITY INTEGRATIONS LIKE '<<integration-name>>'; err <nil> duration 439.877ms args {}]: timestamp=2023-04-25T18:04:59.412+0530
2023-04-25T18:04:59.413+0530 [INFO]  provider.terraform-provider-snowflake_v0.62.0: 2023/04/25 18:04:59 [DEBUG] sql-rows-next [err <nil> duration 1.47175ms]: timestamp=2023-04-25T18:04:59.413+0530
2023-04-25T18:04:59.762+0530 [ERROR] provider.terraform-provider-snowflake_v0.62.0: Response contains error diagnostic: diagnostic_detail= diagnostic_severity=ERROR tf_req_id=e0fdc6f6-25ae-5684-6a5f-b9803f697a4b tf_rpc=ReadResource @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/diag/diagnostics.go:55 @module=sdk.proto diagnostic_summary="error parsing show result: error scanning result: sql: Scan error on column index 4, name "comment": converting NULL to string is unsupported" tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/Snowflake-Labs/snowflake tf_resource_type=snowflake_external_oauth_integration timestamp=2023-04-25T18:04:59.762+0530
```

**Expected behavior**

Null comments should be tolerated when querying the security integration.

**Code samples and commands**

`terraform plan` 

**Additional context**

Removing an existing security integration (external_oauth) resource.

**Comment by merlixo** [comment]
I had the same issue.
When `comment` is not set in the `snowflake_external_oauth_integration` config, then `comment` is set as "" (empty string) in the Terraform state, but as `NULL` in Snowflake. The provider's code does not seems to support this.

To fix this, I had to update `comment` in the Snowflake Security Integration:

`ALTER SECURITY INTEGRATION OKTA_OAUTH SET COMMENT = '';`

**Comment by sfc-gh-ngaberel** [comment]
Hi @rahulj51 @merlixo, thanks for reporting this issue, we're in the middle of a [SDK rewrite](https://github.com/Snowflake-Labs/terraform-provider-snowflake/discussions/1767) and this resource implementation ended up in a weird in-between state, sorry for the inconvenience.
I fixed this issue in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/1900, the fix will be released alongside [v0.67.0](https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/1880), either this week or next.
